pkg/controller/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -1668,6 +1670,97 @@ var _ = Describe("HyperconvergedController", func() {
 
 			})
 		})
+
+		Context("Upgrade Mode", func() {
+			expected := getBasicDeployment()
+			okConds := expected.hco.Status.Conditions
+
+			const (
+				OldVersion = "1.0.0"
+				NewVersion = "1.1.0"
+			)
+
+			BeforeEach(func() {
+				os.Setenv("CONVERSION_CONTAINER", "registry.redhat.io/container-native-virtualization/kubevirt-v2v-conversion:v2.0.0")
+				os.Setenv("VMWARE_CONTAINER", "registry.redhat.io/container-native-virtualization/kubevirt-vmware:v2.0.0}")
+				os.Setenv("OPERATOR_NAMESPACE", namespace)
+			})
+
+			It("Should update HCO Version Id in the CR on init", func() {
+				os.Setenv(util.HcoKvIoVersionName, OldVersion)
+
+				expected.hco.Status.Conditions = nil
+
+				cl := expected.initClient()
+				foundResource := checkAvailability(expected.hco, cl, true, corev1.ConditionFalse)
```

Comment:
basically this is the only place where we call checkAvailability with requeue = true; I'm just wondering if we can change the logic here and keep the function simpler